Report/Metrics.cs

```diff
@@ -0,0 +1,159 @@
+        {
+            if (equityCurve.IsEmpty || orders.Count == 0)
+            {
+                return new Series<Symbol, double>(new Symbol[] { }, new double[] { });
+            }
+
+            var portfolioHoldings = PortfolioLooper.FromOrders(equityCurve, orders)
+                .ToList() // Required because for some reason our AbsoluteHoldingsValue is multiplied by two whenever we GroupBy on the raw IEnumerable
+                .GroupBy(x => x.Time)
+                .Select(kvp => kvp.Last())
+                .ToList();
+
+            var totalPortfolioValueOverTime = (double)portfolioHoldings.Sum(x => x.Holdings.Sum(y => y.AbsoluteHoldingsValue));
+            var holdingsBySymbolOverTime = new Dictionary<Symbol, double>();
+
+            foreach (var portfolio in portfolioHoldings)
+            {
+                foreach (var holding in portfolio.Holdings)
+                {
+                    if (!holdingsBySymbolOverTime.ContainsKey(holding.Symbol))
+                    {
+                        holdingsBySymbolOverTime[holding.Symbol] = (double)holding.AbsoluteHoldingsValue;
+                        continue;
+                    }
+
+                    holdingsBySymbolOverTime[holding.Symbol] = holdingsBySymbolOverTime[holding.Symbol] + (double)holding.AbsoluteHoldingsValue;
+                }
+            }
+
+            return new Series<Symbol, double>(
+                holdingsBySymbolOverTime.Keys,
+                holdingsBySymbolOverTime.Values.Select(x => x / totalPortfolioValueOverTime).ToList()
+            ).DropMissing();
+        }
+
+        /// <summary>
+        /// Strategy long/short exposure by asset class
+        /// </summary>
+        /// <param name="equityCurve">Equity curve</param>
+        /// <param name="orders">Orders of the strategy</param>
+        /// <param name="direction">Long/Short</param>
+        /// <returns>
+        /// DataFrame keyed by <see cref="SecurityType"/> and <see cref="OrderDirection"/>.
+        /// Returns a DataFrame of exposure per asset per direction over time
+        /// </returns>
+        public static Frame<DateTime, Tuple<SecurityType, OrderDirection>> Exposure(Series<DateTime, double> equityCurve, List<Order> orders, OrderDirection direction)
+        {
+            if (equityCurve.IsEmpty || orders.Count == 0)
+            {
+                return Frame.CreateEmpty<DateTime, Tuple<SecurityType, OrderDirection>>();
+            }
+
+            // We want to add all of the holdings by asset class to a mock dataframe that is column keyed by SecurityType with
+            // rows being DateTime and values being the exposure at that given time (as double)
+            var holdingsByAssetClass = new Dictionary<SecurityType, List<KeyValuePair<DateTime, double>>>();
+            var multiplier = direction == OrderDirection.Sell ? -1 : 1;
+            var portfolioLooper = PortfolioLooper.FromOrders(equityCurve, orders);
```

Comment:
Seems we are calling PortfolioLooper.FromOrders multiple times. Couldn't we perform it once and provide it?